start of parser table

rjpepmeyer opened the pull request:
Concerns issue #6

Commit:
start of parser table

dmnemec commented on parser.cpp:
```diff
@@ -1,5 +1,86 @@
+      col.value = "(") return 17;
+      else return 0;
+    case exprprime:
+      //TODO
```
`if (col.value == "<") return 1;`
`else if (col.value == "=") return 2;`

dmnemec replied:
not sure if this is correct yet, as this may not have been entered properly in the parsing table.

rjpepmeyer replied:
I filled out the rest so it's ready for bug fixing now

DMNemec commented on parser.cpp:
```diff
@@ -1,5 +1,86 @@
+      if (col.value == "if" || col.value == "not" || 
+      col.type == identifier || col.type == number ||
+      col.type == boolean || col.value == "-" || 
+      col.value = "(") return 21;
```
should be "return 2" according to the parsing table

Commit:
Fill out parser table

dmnemec committed:
finished implementing parser table

Last 3 values are commented out, as there are no enumeration values for
them yet. When/if they're implemented then they can be uncommented.

May have issues with type/booleanvalue/number types and need to adjust
them

Commit:
start of parser, doesnt work

Commit:
uncomment part of table

Commit:
Make error message cleaner

rpepmeyer committed:
Fix some errors, add debugging

rpepmeyer committed:
Improvements to parser

rpepmeyer committed:
Refactor grammar

rpepmeyer committed:
All full programs pass parser

rpepmeyer committed:
Add error message for garbage at end of file
